improved flow, changed to active voice, improved language (#1)
DerekStrickland/tech-writer-challenge: README.md

## What is the difference between push, pull, and fetch?

> 

Author: How do you feel about adding an introductory paragraph here? I worry that jumping directly into a list without setting the context might be disorienting to users. Plus, you might want to say something that expresses "Here's the cheat sheet in a list, with more detailed explanations below".  

Also, I think it would really be great to explain the difference between a _local_, _tracking_, and _remote_ branch before explaining how to manage each of these using _fetch_ _pull_, and _push_. I had to go review that myself while editing this, so thanks for the refresher!

> - `git fetch` - downloads changes from your _remote_ branch, and adds them to your

Author: I reordered this list and the subsequent paragraphs because _pull_ uses _fetch_ internally, and _push_ comes last in a typical workflow. I "think" that helps a user understand concepts in a stepwise fashion, but let me know if there an specific reason you had it structured differently in case I didn't catch it.

Also, I added emphasis to _local_, _tracking_, and _remote_ here and throughout the rest of the document. I tried to make sure they were consistently and explicitly referenced throughout as well. Implied targets were getting confusing. I hope it was an improvement.  Let me know if you want to work through this together or address this challenge differently.

  _tracking_ branch
- `git pull` - downloads changes from your _remote_ branch, adds them to your
  _tracking_ branch, and then merges them into your _tracking_ branch
- `git push` - uploads changes from your _local_ or _tracking_ branch to your
  _remote_ branch

> Often `git push` and `git pull` are described as equivalent. This isn't entirely

Author: ```suggestion
Often users believe that `git fetch` and `git pull` are equivalent. However, this isn't entirely
```

I think you mean _fetch_ here and not _push_, but wanted you to confirm :smile:

> correct, since `git pull` does two things.

Author: I'm not 💯 percent sure exactly how to add a transition from here to the next section, but I think it needs one. 
Maybe something like this.

```suggestion
accurate, since `git pull` does two things. This tutorial will explain the differences between `git fetch` and `git pull`, and help you understand how they can be used along with `git push` to safely collaborate with your teammates.
```

> The `git fetch` command takes your current branch, and checks to see if there is

Author: I feel like this paragraph needs more "why".  Maybe you could have a discussion about feature branches vs main, `git diff` etc. to help the user know why they want or need to fetch. Whether they are tracking main or a feature branch has big implications with later parts of the workflow.

a _tracking_ branch. If so, it looks for changes in the _remote_ branch. If there
are changes, they are downloaded, but not merged into your _tracking_ branch. You

> must manually issue the `git merge origin/main` command (for the "main" branch)

Author: I changed references from master to main which is our new guideline, but also this section is confusing to me because of the comment above ☝️ .  I think what you are trying to get at is "Here's how to merge the current state of main into your feature branch.", but there are some steps missing if that's the case. Also, what if they are tracking on a shared feature branch?

Finally, this last sentence with the hyphen has some structural issues and should probably be reworded. Maybe we should have a quick zoom so I can better understand what you are going for.

to merge those changes into your branch - probably also called "main".

The `git pull` command performs `git fetch` followed immediately by `git merge`.

> This is often what you want to do, but you may prefer to use `git fetch` followed

Author: Do you think this explanation of why you might want to use git fetch should be part of the previous paragraph? Also, should you spell out explicitly that git pull can make changes and could result in merge conflicts that you have to resolve, etc?  It seems a little off that this command performs two operations, but the paragraph is so lite. Also, it just dawned on me, that you haven't explained what a merge is. Do you think some sort of explanation should be added?

by `git merge` to make sure you understand the changes before you merge.

The `git push` command takes your current branch, and checks to see whether is a

> _tracking_ branch for a _remote_ branch. If so, your changes are pushed from your

Author: Do you think we should explain the whole set upstream concept here if the current branch is only local so far?

_tracking_ branch to your _remote_ branch. This is how code is shared with a

> _remote_ branch. You can think of it as "make the _remote_ branch resemble my

Author: How do you feel about changing this to something like this? I think it helps make it more obvious that it's akin to a db transaction, and therefore should be a really considered action. 
 
```suggestion
_remote_ branch. You can think of it as "update the _remote_ branch with the changes from my
```

_tracking_ branch".

The `git push` command will fail if your _remote_ branch has diverged from your

> _tracking_ branch: if not all the commits in the _remote_ branch are in your

Author: ```suggestion
_tracking_ branch. Your _remote_ branch has _diverged_ if it has changes that are not present in your
```

_tracking_ branch. When this happens, your _tracking_ branch needs to be synchronized

> with the _remote_ branch using either `git pull`, or using `git fetch` with

Author: ```suggestion
with your _remote_ branch using either `git pull`, or using `git fetch` with
```

oops, missed one!
